File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/DiscoverFragment.java
package com.ntsoftware.vspc.myvspc.screens.discover;

import android.os.Bundle;

import androidx.appcompat.app.ActionBar;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.ntsoftware.vspc.myvspc.MainActivity;
import com.ntsoftware.vspc.myvspc.R;
import com.ntsoftware.vspc.myvspc.screens.news.RvNewsAdapter;

import butterknife.BindView;
import butterknife.ButterKnife;

public class DiscoverFragment extends Fragment {

    @BindView(R.id.discover_rv)
    RecyclerView recyclerView;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        View root = inflater.inflate(R.layout.fragment_discover, container, false);

        ButterKnife.bind(this,root);

        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext(), RecyclerView.VERTICAL,false);

        recyclerView.setLayoutManager(linearLayoutManager);



        return  root;
    }

    @Override
    public void onStart() {
        super.onStart();
        recyclerView.setAdapter(new DiscoverAdapter());
    }
}

File: app/src/main/java/com/ntsoftware/vspc/myvspc/screens/discover/DiscoverFragment.kt
package com.ntsoftware.vspc.myvspc.screens.discover

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.ntsoftware.vspc.myvspc.R

class DiscoverFragment: Fragment() {

    private lateinit var recycler_view: RecyclerView

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val root: View = inflater.inflate(R.layout.fragment_discover, container, false)

        recycler_view = root.findViewById(R.id.discover_rv)

        recycler_view.layoutManager = LinearLayoutManager(context,RecyclerView.VERTICAL,false)

        recycler_view.adapter = DiscoverAdapter()

        return root
    }

}